WalletWasabi.Fluent/Models/Wallets/Address.cs

```diff
@@ -0,0 +1,79 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin;
+using ReactiveUI;
+using WalletWasabi.Blockchain.Analysis.Clustering;
+using WalletWasabi.Blockchain.Keys;
+using WalletWasabi.Extensions;
+using WalletWasabi.Hwi;
+using WalletWasabi.Logging;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.Fluent.Models.Wallets;
+
+public class Address : ReactiveObject, IAddress
+{
+	public Address(Wallet wallet, HdPubKey hdPubKey)
+	{
+		Wallet = wallet;
+		HdPubKey = hdPubKey;
+		Network = wallet.Network;
+		HdFingerprint = Wallet.KeyManager.MasterFingerprint;
+		BitcoinAddress = HdPubKey.GetAddress(wallet.Network);
+	}
+
+	public Wallet Wallet { get; }
+	public HdPubKey HdPubKey { get; }
+	public Network Network { get; }
+	public HDFingerprint? HdFingerprint { get; }
+	public BitcoinAddress BitcoinAddress { get; }
+	public SmartLabel Label => HdPubKey.Label;
+	public PubKey PubKey => HdPubKey.PubKey;
+	public KeyPath FullKeyPath => HdPubKey.FullKeyPath;
+	public string Text => BitcoinAddress.ToString();
+	public IEnumerable<string> Labels => Label;
+
+	private bool IsUnused => !Label.IsEmpty && !HdPubKey.IsInternal && HdPubKey.KeyState == KeyState.Clean;
+
+	public bool IsUsed => !IsUnused;
+
+	public void Hide()
+	{
+		Wallet.KeyManager.SetKeyState(KeyState.Locked, HdPubKey);
```

Comment:
?
Locked keys are those reserved for been used in coinjoin outputs.

Reply:
@lontivero this is the current behavior on master:

https://github.com/zkSNACKs/WalletWasabi/blob/master/WalletWasabi.Fluent/ViewModels/Wallets/Receive/ReceiveAddressesViewModel.cs#L145

The logic is untouched, just moved to a different class. If this is wrong then it is wrong on master too.

Reply:
I see. It was introduced in 2021. This is a problem i think. I will open an issue for that.

Reply:
FTR it was introduced in 2020. https://github.com/zkSNACKs/WalletWasabi/issues/10177#issuecomment-1442132898